### PR TITLE
[ART-3477] Fail attach-cve if tracker has no flaw

### PR DESCRIPTION
**Pull request by joepvd**

### CODE REVIEW

File: elliottlib/attach_cve_flaws.py [hunk 1]
```diff
@@ -52,7 +52,19 @@ def get_corresponding_flaw_bugs(bzapi, tracker_bugs, fields):
...
+    flaw_bugs = [flaw_bug for flaw_bug in blocking_bugs if bzutil.is_flaw_bug(flaw_bug)]
+
+    # Validate that each tracker has a corresponding flaw bug
+    flaw_ids = set([b.id for b in flaw_bugs])
+    no_flaws = set()
+    for tracker in tracker_bugs:
+        if len(set(tracker.blocks).intersection(flaw_ids)) == 0:
+            no_flaws.add(tracker.id)
+    if len(no_flaws) > 0:
+        raise exceptions.ElliottFatalError(f'No flaw bugs could be found for these trackers: {no_flaws}')
```
Comment (vfreex) [inline elliottlib/attach_cve_flaws.py:65]: Would like to move this check out of this function. I think I used this function elsewhere so don't really want to break things.

**Review by vfreex — CHANGES_REQUESTED** [review]

File: elliottlib/attach_cve_flaws.py [hunk 2]
```diff
@@ -52,7 +54,22 @@ def get_corresponding_flaw_bugs(bzapi, tracker_bugs, fields):
...
+
+    flaw_bugs = [flaw_bug for flaw_bug in blocking_bugs if bzutil.is_flaw_bug(flaw_bug)]
+
+    # Validate that each tracker has a corresponding flaw bug
+    flaw_ids = set([b.id for b in flaw_bugs])
+    no_flaws = set()
+    for tracker in tracker_bugs:
+        if len(set(tracker.blocks).intersection(flaw_ids)) == 0:
+            no_flaws.add(tracker.id)
+    if len(no_flaws) > 0:
```
Comment (sosiouxme) [inline elliottlib/attach_cve_flaws.py:66]: we have it [on authority from pep 8](https://www.quora.com/What-is-the-Pythonic-way-to-check-for-an-empty-set) not to use `len` for this :)
```suggestion
        if not set(tracker.blocks).intersection(flaw_ids):
            no_flaws.add(tracker.id)
    if no_flaws:
```

**Review by sosiouxme — COMMENTED** [review]
Otherwise LGTM, thanks!

**Review by vfreex — APPROVED** [review]

**Comment by vfreex** [comment]
PR needs rebase

**Pull request merged**